modules/osint/phone_number_search.py

```diff
@@ -0,0 +1,46 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import re
+
+meta = {
+        'name': 'Phone Number Search',
+        'author': 'Kaushik',
+        'version': '0.1',
+        'description': 'A Search Engine for Phone Number Enumeration.',
+        'sources': ('numverify'),
+        'options': (
+                ('number', None, True, 'Phone number (Must include area code)', '-n', 'store', str),
+        ),
+        'examples': ('phone_number_search -q 911234567890')
+}
+
+def module_api(self):
+	num = self.options['number']
+	if re.search(r"^\+\d{1,2}\s?\d{10}$", num):
```

Comment:
remove if-else structure. If the phone is invalid, the API can handle it.

Reply:
You mean i should move it to the engine?

Reply:
How do I tell the user the correct format?
+{area code}{ten digit number}

Reply:
How the API tell you that the input phone is invalid? also any alert and print in module_api won't print.

Reply:
Just a ten digit number without area code is invalid, but it doesn't say what is wrong, so I have to.
![Screenshot 2021-04-04 at 12 07 41 AM](https://user-images.githubusercontent.com/59250093/113488135-c80c5e80-94d9-11eb-9df2-fb1faf35264e.png)

Reply:
self.error works just fine from inside module_api
![Screenshot 2021-04-04 at 12 48 46 AM](https://user-images.githubusercontent.com/59250093/113489061-854d8500-94df-11eb-830b-4dee2e8fa52e.png)